### PR TITLE
Various memleak fixes

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Fix various memory leaks reported by Coverity
- Based on https://github.com/rizinorg/rizin/pull/1891 to get ASAN CI green first

**Test plan**

CI is green

### CODE REVIEW

File: librz/analysis/type_pdb.c [hunk 1]
```diff
@@ -377,13 +378,13 @@ static RzTypeStructMember *parse_struct_member(const RzTypeDB *typedb, RzPdbTpiS
 	}
 	case LF_BCLASS:
 		// For structure, we don't need base class for now
-		return NULL;
+		goto cleanup;
```
Comment (ret2libc) [inline librz/analysis/type_pdb.c:381]: was this really causing any problem? It seems that in these cases the `type` variable remains NULL, thus there is nothing to free.

File: librz/analysis/type_pdb.c [hunk 2]
```diff
@@ -565,6 +567,9 @@ static RzType *parse_union(const RzTypeDB *typedb, RzPdbTpiStream *stream, RzPdb
 	RzType *typ = RZ_NEW0(RzType);
 	if (!typ) {
 		rz_type_base_type_free(base_type);
+		if (to_free_name) {
+			free(name);
+		}
 		return NULL;
```
Comment (ret2libc) [inline librz/analysis/type_pdb.c]:
```suggestion
		base_type = NULL;
		goto cleanup;
```

**Review by ret2libc — COMMENTED** [review]
LGTM apart from that one comment.

File: librz/analysis/type_pdb.c (hunk 1 above)
Comment (XVilka) [reply librz/analysis/type_pdb.c:381]: It wasn't, but I added it for consistency.

Comment (ret2libc) [reply librz/analysis/type_pdb.c:381]: ok

**Review by ret2libc — APPROVED** [review]

**Pull request merged**